examples/disaggregated/slurm/benchmark/submit.py

```
@@ -622,12 +622,22 @@ def submit_job(config, log_dir, dry_run):
-                # start_worker.sh (no hostfile/gpu_map needed).
+                # every rank on that node is given the node's full GPU list and
+                # binds to its own device via mapping.local_rank
+                # (= rank % gpus_per_node). Exposing the whole node is required
+                # for intra-node TP custom all-reduce (attention_dp=false /
+                # TEP), whose cudaDeviceCanAccessPeer() topology check must see
+                # the peer GPUs; pinning one GPU per rank only works for DEP.
                 node_list = list(allocation["nodes"].keys())
                 num_nodes = len(node_list)
+                # Whole-node ownership means every node carries the same GPU
+                # layout, so the first node's list applies to all ranks.
+                gpu_ids = sorted(list(allocation["nodes"].values())[0])
```

Ruff (0.15.21)

[warning] Prefer `next(iter(allocation["nodes"].values()))` over single element slice

Replace with `next(iter(allocation["nodes"].values()))`

(RUF015)